src/nublado2/resourcemgr.py

```
@@ -60,6 +78,86 @@ async def create_user_resources(self, spawner: Spawner) -> None:
+        token = await self._mint_admin_token(base_url, None)
+        mp_ep = f"{base_url}/moneypenny"
+        endpt = f"{mp_ep}/commission"
+        auth = {"Authorization": f"Bearer {token}"}
+        self.log.debug(f"Posting dossier {dossier} to {endpt}")
+        resp = await hc.post(endpt, json=dossier, headers=auth)
+        self.log.debug(f"POST got {resp.status}")
+        resp.raise_for_status()
+        expiry = datetime.datetime.now() + datetime.timedelta(seconds=300)
+        count = 0
+        route = f"{mp_ep}/{uname}"
+        while datetime.datetime.now() < expiry:
+            count += 1
+            self.log.debug(f"Checking Moneypenny status at {route}: #{count}")
+            resp = await hc.get(f"{route}", headers=auth)
+            status = resp.status
+            self.log.debug(f"Moneypenny status: {status}")
+            if status == 200 or 404:
+                return
+            if status != 202:
+                raise RuntimeError(
+                    f"Unexpected status from Moneypenny: {status}"
+                )
+            await asyncio.sleep(int(math.log(count)))
+        raise RuntimeError("Moneypenny timed out")
+
+    async def _make_dossier(
+        self, name: str, auth_state: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        dossier = {
+            "username": name,
+            "uid": int(auth_state["uid"]),
+            "groups": auth_state["groups"],
+        }
+        self.log.debug(f"Dossier: {dossier}")
+        return dossier
+
+    async def _mint_admin_token(
+        self, base_url: str, signing_key_path: Optional[str]
+    ) -> str:
+        """Allowing specification of the signing key makes testing easier."""
+        template_file = os.path.join(
+            os.path.dirname(__file__), "static/moneypenny-jwt-template.json"
+        )
+        with open(template_file, "r") as f:
+            token_template = strTemplate(f.read())
+        if not signing_key_path:
```

Don't check this, just let it throw an exception.  Otherwise I'm very confused about where this comes from.

---

Wait, you're only calling this once, and it's passing in None for the signing key?

Just put the signing key path in the nublado config and expose it as a property and use it like that.

---

Is there an easy way to override that for testing?  That's where I was going with this; I want to pass in a different key when I run a unit test.  Not that there are any tests for this (or any of the other resourcemgr functions) at the moment.

---

I've added a config item in charts, and then in a putative unit test we can pass in the path with our test key.